server/server.go

```diff
@@ -217,7 +230,7 @@ func configureBlockchain(conf *Config, store storage.Store) (*emulator.Blockchai
 		emulator.WithTransactionFeesEnabled(conf.TransactionFeesEnabled),
 	}
 
-	if conf.ServicePublicKey != nil {
+	if conf.ServicePrivateKey == nil && conf.ServicePublicKey != nil {
```

Comment:
Can you help me understand why do you need to change any functionality regarding passing the private keys in order to finish this task? You deploy contracts by using create account functionality which is working with the current master implementation. I'm not saying you shouldn't improve something if you know how just trying to understand better why you want to change this?